### PR TITLE
Make 'send to tab' send to the original tab, if open

### LINKED ISSUE #369: 'send to tab' should send the video to same tab it was open from

### PR DESCRIPTION
**Pull request by vickychijwani**
This fixes #369.

### CODE REVIEW

File: lib/panel-utils.js [hunk 1]
```diff
@@ -324,6 +317,32 @@ function messageHandler(opts) {
...
+    const tabs = require('sdk/tabs');
+    let targetTab = null;
+    for (let tab of tabs) { // eslint-disable-line prefer-const
+      if (tab.id === opts.tabId) {
+        targetTab = tab;
+        break;
+      }
+    }
+    if (targetTab !== null) {
+      targetTab.url = pageUrl;
```
Comment (meandavejustice) [inline lib/panel-utils.js]: this should be `targetTab.activate()`
the above code only refreshes the tab, it doesn't make it active.

**Comment by meandavejustice** [comment]
It would be nice if this also supported the case where a video is launched from an embed or context menu. And then opened from that link.

In other words, it would be good if we could check the actual urls to see if we have the page open, instead of solely storing a reference to the tab.

**Comment by vickychijwani** [comment]
@meandavejustice handled that case now by checking that the video id matches - since the URLs could have slightly different parameters like time.

File: lib/panel-utils.js [hunk 2]
```diff
@@ -324,6 +318,36 @@ function messageHandler(opts) {
...
+      if (tab.id === opts.tabId) {
+        targetTab = tab;
+        break;
+      }
+    }
+
+    const targetVideoId = getVideoId(targetTab.url);
+    const pageVideoId = getVideoId(pageUrl);
+    if (targetTab !== null && pageVideoId && targetVideoId === pageVideoId) {
+      targetTab.url = pageUrl;
```
Comment (vickychijwani) [inline lib/panel-utils.js]: I'm not sure what the intended behaviour is here. Do we want to simply close the min-vid panel and activate the tab, leaving it up to the user to seek to the desired time and play it? Or do we want to auto-play the video from the exact time the panel was closed? I was assuming the latter, which is why I'm setting the URL here.

Comment (meandavejustice) [inline lib/panel-utils.js]: ahh I see, the original intention was to simple activate the already opened tab. You make a good point about continuing playback though. Somehow I wasn't taking that into account. I'll think about it a bit and play with it over the weekend and see what feels best. Thanks!

Comment (meandavejustice) [inline lib/panel-utils.js]: after thinking about it a bit, i think this is the best solution. Thank you @vickychijwani

**Comment by vickychijwani** [comment]
Rebased and resolved the merge conflicts.

**Pull request merged**